backend/recipe/Recipe.py

```diff
@@ -31,8 +31,13 @@ def addRecipe(id, recipes):
             result = cursor.fetchone()
             current_recipes_json = result[0] if result else '[]'
             current_recipes = json.loads(current_recipes_json)
+            recipe_exists = False
+            for rec in current_recipes:
+                if rec['name'] == recipes[0].name:
+                    recipe_exists = True
 
-            current_recipes.extend(recipes)
+            if not recipe_exists:
+                current_recipes.extend(recipes)
 
             updated_recipes_json = json.dumps(current_recipes, cls=RecipeEncoder)
             cursor.execute('UPDATE recipes SET recipe_lists = ? WHERE id = ?', (updated_recipes_json, id))
```

Comment:
I believe that id is now usr_id in the updated DB, all tables now have "usr_id" rather that "id"